### PR TITLE
[KernelGen][Nvidia] Add rrelu operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `rrelu`. Applies the randomized leaky ReLU (RReLU) function element-wise. In training
mode the slope for negative inputs is drawn uniformly from [lower, upper],
while in evaluation mode the average of lower and upper is used.

## Testing
- Parametrized tests over multiple shapes and float16/float32/bfloat16 dtypes (30 cases)
- Validated against PyTorch reference on device side
- Tested on: Nvidia

## Repository / Branch
- Target repository: `flagos-ai/FlagGems`
- Base branch: `master`

## Performance
Test command: `pytest benchmark/test_rrelu.py --level core` (NVIDIA H20)

### float16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 1.100 | 1.270 | 0.866 |
| 64, 64 | 0.006 | 0.006 | 1.006 |
| 4096, 4096 | 0.024 | 0.028 | 0.845 |
| 64, 512, 512 | 0.024 | 0.028 | 0.848 |
| 1024, 1024, 1024 | 1.160 | 1.330 | 0.872 |
| 1073741824 | 2.089 | 2.195 | 0.952 |
| 64, 64 | 0.006 | 0.006 | 1.074 |
| 4096, 4096 | 0.038 | 0.040 | 0.942 |
| 64, 512, 512 | 0.037 | 0.040 | 0.925 |
| 1024, 1024, 1024 | 2.109 | 2.182 | 0.966 |

### float32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 1.129 | 1.263 | 0.894 |
| 64, 64 | 0.006 | 0.006 | 1.063 |
| 4096, 4096 | 0.024 | 0.028 | 0.836 |
| 64, 512, 512 | 0.024 | 0.028 | 0.852 |
| 1024, 1024, 1024 | 1.152 | 1.292 | 0.891 |
| 1073741824 | 1.210 | 1.278 | 0.946 |
| 64, 64 | 0.006 | 0.006 | 0.983 |
| 4096, 4096 | 0.023 | 0.028 | 0.821 |
| 64, 512, 512 | 0.023 | 0.028 | 0.821 |
| 1024, 1024, 1024 | 1.100 | 1.302 | 0.845 |

### bfloat16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 2.016 | 2.190 | 0.921 |
| 64, 64 | 0.006 | 0.006 | 1.003 |
| 4096, 4096 | 0.037 | 0.039 | 0.956 |
| 64, 512, 512 | 0.037 | 0.039 | 0.955 |
| 1024, 1024, 1024 | 2.052 | 2.172 | 0.945 |
| 1073741824 | 1.150 | 1.285 | 0.895 |
| 64, 64 | 0.006 | 0.006 | 1.000 |
| 4096, 4096 | 0.023 | 0.027 | 0.826 |
| 64, 512, 512 | 0.023 | 0.028 | 0.823 |
| 1024, 1024, 1024 | 1.125 | 1.292 | 0.871 |

**Arithmetic Mean Speedup: 0.915x**
## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (30 cases, --level core) | 0.915 | Primary |
| Tianshu | PASS | PASS (9 cases) | 1.03 | — |
| Muxi | PASS | PASS (9 cases) | 2.19 | — |
| Ascend | PASS | — | — | — |
| Hygon | FAIL | FAIL | — | Error: '_OpNamespace' 'aten' object has no attribute 'rrelu_with_noise_functiona... |

## Files Changed
- `src/flag_gems/ops/rrelu.py`: Triton kernel implementation
- `tests/test_rrelu.py`: Accuracy test
- `benchmark/test_rrelu.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.4)

### CODE REVIEW

File: conf/operators.yaml [hunk 1]
```diff
@@ -13007,6 +13007,31 @@ ops:
...
+    while in evaluation mode the average of lower and upper is used.
+  for:
+  - rrelu
+  labels:
+  - aten
+  - KernelGen
+  kind:
+  - NeuralNetwork
+  stages:
+  - alpha: '5.4'
```
Comment (bin913) [inline conf/operators.yaml]:
```suggestion
  - alpha: '5.5.0'
```

File: conf/operators.yaml [hunk 2]
```diff
@@ -13007,6 +13007,31 @@ ops:
...
+  description: The in-place version of `rrelu()`.
+  for:
+  - rrelu_
+  labels:
+  - aten
+  - KernelGen
+  kind:
+  - Math
+  stages:
+  - alpha: '5.4'
```
Comment (bin913) [inline conf/operators.yaml]:
```suggestion
  - alpha: '5.5.0'
```

File: src/flag_gems/ops/rrelu.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        # transform itself is computed by the Triton kernel below.
+        noise = torch.rand_like(self) * (upper - lower) + lower
+        return rrelu_with_noise_forward(self, noise)
+    else:
+        slope = (lower + upper) * 0.5
+        return rrelu_forward(self, slope)
+
+
+def rrelu_(
+    self, lower=0.125, upper=0.33333333333333331, training=False, generator=None
```
Comment (bin913) [inline src/flag_gems/ops/rrelu.py:62]: Why are lower and upper defaulted to these two values?

File: src/flag_gems/ops/rrelu.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    nf = noise.to(tl.float32)
+    return tl.where(xf >= 0, xf, xf * nf).to(x.dtype)
+
+
+def _check_args(lower, upper, generator):
+    assert generator is None, "generator is not supported in FlagGems"
+    assert lower <= upper, "lower must be no greater than upper"
+
+
+def rrelu(self, lower=0.125, upper=0.33333333333333331, training=False, generator=None):
```
Comment (bin913) [inline src/flag_gems/ops/rrelu.py:48]: Why are lower and upper defaulted to these two values?
